src/fsharp/FSharp.Core/list.fs

```diff
@@ -472,10 +472,11 @@ namespace Microsoft.FSharp.Collections
         let skip count list =
             if count <= 0 then list else
             let rec loop i lst =
+                if i = 0 then lst else
                 match lst with
-                | _ when i = 0 -> lst
                 | _ :: t -> loop (i-1) t
                 | [] -> invalidArgOutOfRange "count" count "distance past the list" i
+
```

Comment:
This should be a separate formatting PR since it's not inherent to the purpose of this one

Reply:
done